src/EventStore.Core/ProcessTracker.cs

```diff
@@ -0,0 +1,190 @@
+	}
+
+	public long CollectFreeMemoryUnix() {
+		return Bookkeeping(() => (long)_stats.GetFreeMemory());
+	}
+
+	public long CollectFreeMemoryWindows() {
+		return Bookkeeping(() => _perfCounter.GetFreeMemory());
+	}
+
+	public long CollectTotalMemoryUnix() {
+		return Bookkeeping(() => (long)_stats.GetTotalMemory());
+	}
+
+	public long CollectTotalMemoryWindows() {
+		return Bookkeeping(() => (long)WinNativeMemoryStatus.GetTotalMemory());
+	}
+
+	public Measurement<double> CollectProcCPU() {
+		return Bookkeeping(() => {
+			var dim = new KeyValuePair<string, object?>[] {
+				new("id", _process!.Id),
+				new("mem", _process!.WorkingSet64),
+				new("startTime", _process!.StartTime.ToUniversalTime().ToString("O")),
+				new("threadsCount", ThreadPool.ThreadCount),
+				new("contentionRate", Monitor.LockContentionCount),
+				new("thrownExceptionsRate", _eventCountersHelper.GetThrownExceptionsRate()),
+			};
+
+			return new Measurement<double>(ProcCpuUsage(), dim);
+		});
+	}
+
+	public long CollectDriveIOReadBytes() {
+		return (long)Bookkeeping(() => _diskIo?.ReadBytes ?? 0);
+	}
+
+	public long CollectDriveIOReadOps() {
+		return (long)Bookkeeping(() => _diskIo?.ReadOps ?? 0);
+	}
+
+	public long CollectDriveIOWriteBytes() {
+		return (long)Bookkeeping(() => _diskIo?.WrittenBytes ?? 0);
+	}
+
+	public long CollectDriveIOWriteOps() {
+		return (long)Bookkeeping(() => _diskIo?.WriteOps ?? 0);
+	}
+
+	public long CollectDriveTotalBytes() {
+		return Bookkeeping(() => _esDriveInfo?.TotalBytes ?? 0);
+	}
+
+	public long CollectDriveUsedBytes() {
+		return Bookkeeping(() => _esDriveInfo?.UsedBytes ?? 0);
+	}
+
+	public long CollectGcAllocationSpeed() {
+		return Bookkeeping(() => GC.GetTotalAllocatedBytes());
+	}
+
+	public double CollectGcFragmentation() {
+		return Bookkeeping(() => {
+			var info = GC.GetGCMemoryInfo();
+			return info.HeapSizeBytes != 0 ? info.FragmentedBytes * 100d / info.HeapSizeBytes : 0;
+		});
+	}
+
+	public long CollectGc0Size() {
+		return Bookkeeping(() => _eventCountersHelper.GetGcStats().Gen0Size);
+	}
+
+	public long CollectGc1Size() {
+		return Bookkeeping(() => _eventCountersHelper.GetGcStats().Gen1Size);
+	}
+
+	public long CollectGc2Size() {
+		return Bookkeeping(() => _eventCountersHelper.GetGcStats().Gen2Size);
+	}
+
+	public long CollectGcLohSize() {
+		return Bookkeeping(() => _eventCountersHelper.GetGcStats().LargeHeapSize);
+	}
+
+	public int CollectGc0Count() {
+		return Bookkeeping(() => GC.CollectionCount(0));
+	}
+
+	public int CollectGc1Count() {
+		return Bookkeeping(() => GC.CollectionCount(1));
+	}
+
+	public int CollectGc2Count() {
+		return Bookkeeping(() => GC.CollectionCount(2));
+	}
+
+	public float CollectGcTimeInGc() {
+		return Bookkeeping(() => _eventCountersHelper.GetGcStats().TimeInGc);
+	}
+
+	public long CollectGcTotalBytesInHeap() {
+		return Bookkeeping(() => GC.GetGCMemoryInfo().HeapSizeBytes);
+	}
+
+	double ProcCpuUsage() {
```

Comment:
i think this might be calculating average cpu usage since the process started

Reply:
I got rid of it